### ISSUE TITLE
Add Ranker feature support

### ISSUE THREAD

**Issue by tkhshtsh0917** [soda-inria/tabicl#43]
Hi, team.
First, I would like to express my heartfelt gratitude for sharing this amazing foundation model!

Would you consider adding LTR (Learning to Rank) feature support?
Various gradient boosting tree libraries (`xgboost`, `lightgbm`, `catboost`) support `Ranker` feature alongside `Classifier` and `Regressor`. In production information retrieval systems, ranking models trained using these libraries are still widely used.

However, I believe that the TabICLv2 you invented the capability to interpret tabular data at a more advanced level, making it highly likely to achieve a new breakthrough in LTR as well.
More specifically, I believe promising results could be achieved by building upon the `TabICLRegressor` and adding a linear layer, then training it using a listwise ranking loss (e.g. [NeuralNDCG](https://arxiv.org/abs/2102.07831) etc.) 🤔

Let me know if you're interested! ;-)
